Mage.Sets/src/mage/cards/i/IceCave.java

```diff
@@ -0,0 +1,116 @@
+ *  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ *  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *  The views and conclusions contained in the software and documentation are those of the
+ *  authors and should not be interpreted as representing official policies, either expressed
+ *  or implied, of BetaSteward_at_googlemail.com.
+ */
+package mage.cards.i;
+
+import mage.abilities.Ability;
+import mage.abilities.common.SpellCastAllTriggeredAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.cards.CardsImpl;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.SetTargetPointer;
+import mage.constants.Zone;
+import mage.filter.FilterSpell;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.game.stack.Spell;
+import mage.players.Player;
+import mage.util.CardUtil;
+
+import java.util.UUID;
+import mage.abilities.costs.Cost;
+import mage.abilities.costs.mana.ManaCostsImpl;
+
+/**
+ *
+ * @author ThomasLerner
+ */
+public class IceCave extends CardImpl {
+
+    public IceCave(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId,setInfo,new CardType[]{CardType.ENCHANTMENT},"{3}{U}{U}");
+
+
+        // Whenever a player casts a spell, any other player may pay that spell's mana cost. If a player does, counter the spell. (Mana cost includes color.)
+        this.addAbility(new SpellCastAllTriggeredAbility(Zone.BATTLEFIELD, new IceCaveEffect(), new FilterSpell(), false, SetTargetPointer.SPELL));
+    }
+
+    public IceCave(final IceCave card) {
+        super(card);
+    }
+
+    @Override
+    public IceCave copy() {
+        return new IceCave(this);
+    }
+}
+
+class IceCaveEffect extends OneShotEffect {
+
+    public IceCaveEffect() {
+        super(Outcome.AIDontUseIt);
+        this.staticText = "any other player may pay that spell's mana cost. If a player does, counter the spell";
+    }
+
+    public IceCaveEffect(final IceCaveEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public IceCaveEffect copy() {
+        return new IceCaveEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        Permanent sourcePermanent = game.getPermanentOrLKIBattlefield(source.getSourceId());
+        Spell spell = (Spell) game.getStack().getStackObject(targetPointer.getFirst(game, source));
+        if(sourcePermanent != null && spell != null) {
+            Cost cost = new ManaCostsImpl(spell.getSpellAbility().getManaCosts().getText());
+            for (UUID playerId : game.getState().getPlayerList(source.getControllerId())) {
+                Player player = game.getPlayer(playerId);
+                if(player.getId() != spell.getControllerId()) {
```

Comment:
GetOpponents is not correct, because other players can include non opponents. We don't have a format yet with non opponent other players, but it will take place in the future.
So best use getPlayersInRange(source.getControllerId() for the loop and ommit the source controller.